Спринт №15

Keithcutt opened the pull request:
Код экрана плеера разнесен по пакетам presentation, data и domain. 
Связь между ними организована посредством интерфейсов.

Keithcutt committed:
Начало работы над домащним заданием спринта №15.

Keithcutt committed:
Продолжение работы над домащним заданием спринта №15.

Keithcutt committed:
Продолжение работы над домашним заданием спринта №15.

Keithcutt committed:
Продолжение работы над домашним заданием спринта №15.

Keithcutt committed:
Продолжение работы над домашним заданием спринта №15.

Keithcutt committed:
Продолжение работы над домашним заданием спринта №15.

Keithcutt committed:
Продолжение работы над домашним заданием спринта №15.

Keithcutt committed:
Код PlayerActivity был разнесен по пакетам presentation и data.
Организована связь между пакетами посредством интерфейсов.

Grigoriy-M commented on app/src/main/java/com/example/playlistmaker/data/repository/PlayerRepositoryImpl.kt:
```diff
@@ -0,0 +1,55 @@
+class PlayerRepositoryImpl() : PlayerRepository {
+
+    private var playerState = PlayerState.DEFAULT
+    private var mediaPlayer = MediaPlayer()
```
Передавай MediaPlayer() в конструктор как зависимость.

Grigoriy-M commented on app/src/main/java/com/example/playlistmaker/domain/repository/OnPlayerStateChangeListener.kt:
```diff
@@ -0,0 +1,7 @@
+
+import com.example.playlistmaker.domain.models.PlayerState
+
+interface OnPlayerStateChangeListener {
```
замени на "fun interface OnPlayerStateChangeListener"
Тогда можно сократишь код на реализации, например так.
playerInteractor.setOnPlayerStateChangeListener(::playbackCases)

Grigoriy-M commented on app/src/main/java/com/example/playlistmaker/ui/player/PlayerActivity.kt:
```diff
@@ -21,50 +23,54 @@ class PlayerActivity : AppCompatActivity() {
         binding.backButton.setOnClickListener { finish() }
 
-        preparePlayer(currentTrack.previewUrl)
+        playerInteractor.setOnPlayerStateChangeListener(object : OnPlayerStateChangeListener {
```
замени на "fun interface OnPlayerStateChangeListener"
Тогда можно сократишь код на реализации, например так.
playerInteractor.setOnPlayerStateChangeListener(::playbackCases)

Grigoriy-M commented on app/src/main/java/com/example/playlistmaker/ui/player/PlayerActivity.kt:
```diff
@@ -21,50 +23,54 @@ class PlayerActivity : AppCompatActivity() {
 
+    private fun getCurrentTrack(): Track {
+        val json = intent.getStringExtra(TRACK_KEY)
+        return Gson().fromJson(json, Track::class.java)
```
Gson() лучше получать как зависимость из Creator

Keithcutt committed:
Исправление ошибок после ревью

1. Добавлен скролл на экран плеера.
2. Теперь MediaPlayer передается в PlayerRepository как зависимость.
3. OnPlayerStateChangeListener теперь функциональный интерфейс, что позволило передавать лямбду в методе setOnPlayerStateChangeListener().
4. Получение текущего трека в PlayerActivity реализовано через Creator.
